openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue

```diff
@@ -242,14 +242,24 @@ export default {
   created() {
     this.internalDisabled = true
     this.api = new OpenC3Api()
-    this.api.get_target_names().then((result) => {
+    this.api.get_target_names().then(async (result) => {
       this.targetNames = result.flatMap((target) => {
         // Ignore the UNKNOWN target as it doesn't make sense to select this
         if (target == 'UNKNOWN') {
           return []
         }
         return { label: target, value: target }
       })
+      // Filter out any targets without packets
+      for (var i = this.targetNames.length - 1; i >= 0; i--) {
+        const cmd =
+          this.mode === 'tlm' ? 'get_all_tlm_names' : 'get_all_cmd_names'
+        await this.api[cmd](this.targetNames[i].value).then((names) => {
```

Comment:
I don't love having to request the names but it's in the created() method and it's the only way to determine if the target has packets.

Reply:
This is too heavy with lots of API calls for lots of targets.

Reply:
Make this a todo and note we can't do lots of API calls.